File: daily/20200503/example_metashape/10union.go
package main

type Node struct {
	Kind string `json:"kind"`
	Atom *Atom `json:"atom"`
	Composite *Composite `json:"composite"`
}


type Atom struct {
	Type string `json:"type"`
}

type Composite struct {
	Type string `json:"type"`
	Args []Node `json:"args"`
}

File: daily/20200503/example_metashape/10union.py
from __future__ import annotations
import typing as t
from emit import emit
from prestring.go.gofmt import gofmt


class Atom:
    type_: str


class Composite:
    type_: str
    args: t.List[Node]


Node = t.Union[Atom, Composite]
Node.__name__ = "Node"
print(gofmt(emit([Node]), always=False))
